### ISSUE TITLE
unoptimizable-recursion misses self-recursion written by plain name (Φ.<name>)

### ISSUE THREAD

**Issue by VasilevNStas** [objectionary/lints#1381]
## Problem

`unoptimizable-recursion` misses the most common way to write a self-recursive call — by plain name — because the modern parser resolves such calls to `Φ.<name>`, while the lint only looks for `ξ.ρ.<name>` (the `^`-style form).

## Root cause

`src/main/resources/org/eolang/lints/design/unoptimizable-recursion.xsl:64-66`:

```xml
<xsl:variable name="self" select="concat('ξ.ρ.', @name)"/>
<xsl:variable name="calls" select="... //o[@base = $self] ..."/>
```

Verified with a real program:

```eo
[] > calc
  if. > @
    n.eq 0
    1
    n.mul (calc (n.sub 1))
```

The inner call `calc (...)` is resolved by eo-parser 0.63.0 to `@base='Φ.calc'`, so the lint finds no `ξ.ρ.calc` and reports nothing. The same recursion written as `^.calc` *does* produce `ξ.ρ.calc` and is caught — so the lint's behavior depends on how the recursion is spelled, not on the recursion itself.

## Expected behavior

Self-calls should be detected for both `Φ.<name>` and `ξ.ρ.<name>` forms (and ideally `$<name>`), so the lint is consistent regardless of the referencing syntax.

## Related

- #1321 (compiler fails to convert certain self-recursion shapes into Java loops) — the underlying design issue.
- `packs/single/unoptimizable-recursion/` — existing packs use the `^` form only.

**Comment by 0crat** [comment]
@vasilevnstas Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+794](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
